presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayElementAtFunction.java

```diff
@@ -106,16 +122,21 @@ public static Block blockElementAt(@TypeParameter("E") Type elementType, @SqlTyp
         return (Block) elementType.getObject(array, position);
     }
 
-    private static int checkedIndexToBlockPosition(Block block, long index)
+    private static boolean indexInRange(Block block, long index)
```

Comment:
This method almost duplicates checkedIndexToBlockPosition. I think it'd be cleaner make the existing:
1. throw if the index is 0
2. return a marker value (e.g., -1) if abs(index) is outside of range [1, length]
3. return the block position otherwise

Callers would then check if the result value is -1 and return null in that case.